src/irmin-pack/unix/sparse_file.ml

```diff
@@ -285,6 +285,13 @@ module Make (Io : Io.S) = struct
   type t = { mapping : Mapping_file.t; data : Io.t }
 
   let v ~mapping ~data = { mapping; data }
+
+  let open_ro ~mapping ~data =
+    let open Result_syntax in
+    let* mapping = Mapping_file.open_map ~path:mapping in
+    let+ data = Io.open_ ~path:data ~readonly:true in
+    { mapping; data }
```

Comment:
I view `v` as the canonical constructor of the type so would prefer this to be `v ~mapping ~data`.

Reply:
Ha thanks, I missed it! (but the `v` is hopefully going away soon :-°)